packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -1366,11 +1366,6 @@ function applyAttributes(ctx: ParserCtx, parsedAttr: ParsedAttribute, element: B
             }
         }
 
-        // Prevent usage of the slot attribute with expression.
-        if (name === 'slot' && ast.isExpression(attr.value)) {
-            ctx.throwOnNode(ParserDiagnostics.SLOT_ATTRIBUTE_CANNOT_BE_EXPRESSION, attr);
```

Comment:
I know that you are still in the middle of your POC. But make sure that you delete the unused diagnostics. 😉